### PR TITLE
python310Packages.pyg4ometry: init at 1.0.2

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
On nixos x86_64:

```
OSError: /nix/store/n8q18ymykjlghni9j9flw3y630w8s0al-gcc-9.5.0-lib/lib/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /nix/store/3zwmn2nsm66n5qmdczdh117mh2ks9c2q-python3.10-pyg4ometry-1.0.2/lib/python3.10/site-packages/pyg4ometry/pycgal/pyg4_cgal.cpython-310-x86_64-linux-gnu.so)
```

File: pkgs/development/python-modules/pyg4ometry/default.nix [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+buildPythonPackage rec {
+  pname = "pyg4ometry";
+  version = "1.0.2";
+  format = "pyproject";
+
+  src = fetchFromBitbucket {
+    owner = "jairhul";
+    repo = pname;
+    rev = "v${version}";
```
Comment (pbsds) [inline pkgs/development/python-modules/pyg4ometry/default.nix:33]:
```suggestion
    rev = "refs/tags/v${version}";
```

**Review by pbsds — COMMENTED** [review]
Heads up: this PR needs a rebase.

File: pkgs/development/python-modules/pyg4ometry/default.nix [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      --replace "/opt/local" "/nowhere" \
+      --replace "antlr4-python3-runtime==4.7.1" "antlr4-python3-runtime" \
+      --replace '"vtk",' "" # requires vtk_9 with VTK_WHEEL_BUILD=ON
+  '';
+
+  nativeBuildInputs = [
+    gitpython
+    cython
+    pypandoc
+    testtools
```
Comment (pbsds) [inline pkgs/development/python-modules/pyg4ometry/default.nix:55]: Can this be moved to `nativeCheckInputs`?

File: pkgs/development/python-modules/pyg4ometry/default.nix [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  # paraview requires paraview
+  preCheck = ''
+    substituteInPlace pyg4ometry/test/runTests.py \
+      --replace "from gdml                   import GdmlLoadTests" "" \
+      --replace "from paraviewExport         import VtkExporter" ""
+  '';
+  checkPhase = ''
+    runHook preCheck
+
+    ${python.interpreter} pyg4ometry/test/runTests.py
```
Comment (pbsds) [inline pkgs/development/python-modules/pyg4ometry/default.nix:89]: A comment about why pytestCheckHook is not used is well appreciated. I'd guess something along the lines of "the unittest tests are not well formed for pytest."

**Review by pbsds — CHANGES_REQUESTED** [review]
##### Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [ ] package build on x86_64-linux
- [ ] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches that are remotely available are fetched with `fetchpatch`

---

Result of `nixpkgs-review pr 188437` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.pyg4ometry</li>
    <li>python310Packages.pyg4ometry.dist</li>
    <li>python311Packages.pyg4ometry</li>
    <li>python311Packages.pyg4ometry.dist</li>
  </ul>
</details>

build logs: https://gist.github.com/pbsds/fb0f1f2790250184057d17b2a4627de8

**Pull request closed without merging**